sw/host/cryptotest/testvectors/parsers/nist_cavp_ecdh_parser.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Parser for converting NIST CAVP ECDH test vectors to JSON.
+
+"""
+
+import argparse
+import sys
+import json
+import jsonschema
+
+from cryptotest_util import parse_rsp
+
+
+# Mapping from the curve names used by NIST to those used by cryptotest
+EC_NAME_MAPPING = {
+    "P-256": "p256",
+    # TODO uncomment when ECDH supports P-384
+    # "P-384": "p384",
+}
+
+
+def parse_testcases(args) -> None:
+    raw_testcases = parse_rsp(args.src)
+    test_cases = list()
+
+    # NIST splits the rsp files into sections with named after the ECC curve.
+    for section_name in raw_testcases.keys():
+        if section_name not in EC_NAME_MAPPING.keys():
+            continue
+        for test_vec in raw_testcases[section_name]:
+            test_case = {
+                "vendor": "nist",
+                "test_case_id": int(test_vec["COUNT"]),
+                "algorithm": "ecdh",
+                "curve": EC_NAME_MAPPING[section_name],
+                # 'CAVS' here stands for the NIST Cryptographic Algorithm
+                # Validation System. In other words, the test harness is
+                # serving as the other peer in the shared key derivation, and
+                # the IUT needs the other peer's public key.
+                "qx": list(bytes.fromhex(test_vec["QCAVSx"])),
+                "qy": list(bytes.fromhex(test_vec["QCAVSy"])),
+                # The wycheproof vectors use "signed hexadecimal notation" for
```

Comment:
Is it possible to enforce this notation in the schema and also include this comment about "signed" notation there to avoid this implicit agreement between the parsers?

Reply:
I don't think so, unfortunately. Either a string or byte array type in the schema would give no indication of the integer format, and using an `int` would lose precision to that of an `f64`.

Reply:
I'll add a comment in the JSON schema but I think we're out of luck when it comes to statically enforcing the format.

Reply:
Could we instead resolve the disagreement by interpreting the wycheproof vectors as unsigned integers? There's no reason `d` should ever be negative, and the cryptolib interprets it as unsigned.

Reply:
I had this thought a few days ago. I can refactor the tests to do this.

Reply:
It turns out Rust's `BigUint` type has no problem ingesting the occasional leading zero that shows up in wycheproof's notation, so we don't have to do anything special in either parser. I only didn't do this before because I didn't know that `d` was guaranteed to be positive.